### ISSUE TITLE
Handle case where `Content-Length` header is not supplied

### ISSUE THREAD

**Issue by zimeon** [resync/resync#41]
Example error:
```
simeon@RottenApple resync> resync-sync -v --capability-list https://pod.stanford.edu/.well-known/resourcesync/normalized-capabilitylist/marcxml --access-token $ACCESS_TOKEN -b  https://pod.stanford.edu/ pod
Reading capability list https://pod.stanford.edu/.well-known/resourcesync/normalized-capabilitylist/marcxml
Reading resource list https://pod.stanford.edu/organizations/normalized_resourcelist/marcxml
Traceback (most recent call last):
  File "/Users/simeon/.python_venv/py38/lib/python3.8/site-packages/resync/client.py", line 101, in read_resource_list
    resource_list.read(uri=uri)
  File "/Users/simeon/.python_venv/py38/lib/python3.8/site-packages/resync/list_base_with_index.py", line 99, in read
    self.content_length = int(fh.info()['Content-Length'])
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

**Comment by zimeon** [comment]
Code was intended to handle this gracefully, but doesn't. In resync/list_base_with_index.py line 99:
```
self.content_length = int(fh.info()['Content-Length'])
```
a missing Content-Lenght doesn't end up generating a `KeyError` but instead a `TypeError` from the `int()`. Should change to catch both

**Comment by zimeon** [comment]
Released in 2.0.1
